src/ml_tooling/baseclass.py

```diff
@@ -198,6 +196,30 @@ def make_prediction(self, input_data, proba=False, use_index=False) -> pd.DataFr
             message = f"You haven't fitted the model. Call 'train_model' or 'score_model' first"
             raise MLToolingError(message) from None
 
+    @property
+    def default_metric(self):
+        """
+        Finds estimator_type for estimator in a BaseClassModel and returns default
+        metric for this class stated in .config. If passed estimator is a Pipeline,
+        assume last step is the estimator.
+
+        Returns
+        -------
+        str
+            Name of the metric
+
+        """
+
+        return self.config.CLASSIFIER_METRIC if self.model._estimator_type == 'classifier' \
+            else self.config.REGRESSION_METRIC
+
+    @default_metric.setter
+    def default_metric(self, metric):
+        if self.model._estimator_type == 'classifier':
+            self.__class__.config.CLASSIFIER_METRIC = metric
```

Comment:
Do we need the `__class__.config` to make this work? Usually we would use the `self.config = metric` syntax

Reply:
Wouldn't this specifiy an instance-specific self.config  which is what we are trying to avoid?

Reply:
```python
class my_class:
    var1 = 1
    var2 = 1
    def set_var1(self):
        self.var1 = 2
    def set_var2(self):
        self.__class__.var2 = 2

cl1 = my_class()
cl2 = my_class()
cl1.set_var1()
cl1.set_var2()
print(cl2.var1)
print(cl2.var2)

```
```
Output:
1
2
```
Small example of my point

Reply:
yes, you are correct - forgot we now have the black magic ConfigGetter 😄